### PR TITLE
[11.0] migrate sentry

### PR DESCRIPTION
**Pull request by njeudy**
- Migrate sentry module to 11.0

### CODE REVIEW

**Comment by njeudy** [comment]
@Yajo how to add raven in ci default install ..? check failed because raven is not installed.

**Comment by yajo** [comment]
Add a requirements.txt file at repo root.

**Comment by LoisRForgeFlow** [comment]
shouldn't this be proposed in one of the new repositories? Not in the [list](https://docs.google.com/spreadsheets/d/1Xg95cW4TFMf_Lo5i_CZC_qOOfN8RgxPRc0LJTLTkdUI/edit?usp=sharing) though...

**Comment by njeudy** [comment]
@Yajo thanks for the hint
@lreficent I think it should stay in server-tools (not ux nor auth) so I just keep it there .. 

will squash commit after first green in travis :)

**Comment by njeudy** [comment]
@lreficent @Yajo merge and test done :)

**Review by yajo — CHANGES_REQUESTED** [review]
Code looks good, but you must keep git history: https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-11.0

**Comment by njeudy** [comment]
@Yajo OCA CLAbot says no cla for Naglis Jonaitis ...

But keep git history ..

**Comment by njeudy** [comment]
In lastest raven lib, FIELDS -> KEYS so update test with this :)

**Comment by njeudy** [comment]
@Yajo ok requested changes done

**Comment by njeudy** [comment]
@Yajo Ok now with git history ?

**Review by yajo — APPROVED** [review]

File: sentry/logutils.py [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+
+class SanitizeOdooCookiesProcessor(SanitizePasswordsProcessor):
+    '''
+    Custom :class:`raven.processors.Processor`.
+
+    Allows to sanitize sensitive Odoo cookies, namely the "session_id" cookie.
+    '''
+
+    KEYS = frozenset([
```
Comment (lasley) [inline sentry/logutils.py]: Can we do [like this](https://github.com/OCA/server-tools/pull/1117/files#diff-e96b25e7c55fcc00203e33cfdc93d8e4R107) instead? This way we support both the old and new libraries. Seeing as it's a new change upstream, I think it's a good move for a year or two.

Comment (njeudy) [reply sentry/logutils.py]: Will check this thanks !!

**Review by lasley — APPROVED** [review]
Thanks @njeudy (assuming CI passes)

**Comment by njeudy** [comment]
@Yajo @lasley who is able to merge ?

**Comment by lasley** [comment]
@njeudy - with two reviews, five days are required for merge. This will be eligible tomorrow if you want to ping again

**Comment by lasley** [comment]
@njeudy - I was about to merge and noticed we have a cross-merge commit in here 662d531. Were you solving a conflict or something?

**Comment by njeudy** [comment]
Yes I have a merge conflict in requirement.txt. did I make a mistake .. I should have rebase and not merge ?

**Comment by njeudy** [comment]
@lasley : just rebase from V11 think it will be ok

**Comment by lasley** [comment]
@njeudy - Looks much better, thank you!

Regarding rebase vs. merge - the rebase is cleaner because we don't end up with another commit.

**Comment by fractalf** [comment]
Is this supposed to work now?
It doesn't work for me using Odoo 11 in the latest docker image

This is the error message I get when trying to install the module
```
Unable to install module "sentry" because an external dependency is not met: No module named raven
```

raven is installed with pip

```
odoo11 # docker exec -it -u root odoo11_odoo_1 /bin/bash
root@2c8be04d88ed:/# pip install raven --upgrade
Requirement already up-to-date: raven in /usr/local/lib/python2.7/dist-packages (6.9.0)
Requirement already satisfied, skipping upgrade: contextlib2; python_version < "3.2" in /usr/local/lib/python2.7/dist-packages (from raven) (0.5.5)
```

odoo.conf:
```
[options]
addons_path = /mnt/extra-addons
data_dir = /var/lib/odoo
server_wide_modules = web
```

\_\_manifest__.py:
```
    'external_dependencies': {
        'python': [
            'raven',
        ]
    },
```

@njeudy
Is this merged with the main branch? Cause I can't get it working.

**Pull request merged**